dwave/optimization/symbols.pyx

```diff
@@ -1554,40 +1534,71 @@ cdef class ListVariable(ArraySymbol):
         # The validity of the state is checked in C++
         self.ptr.initialize_state(self.model.states._states[index], move(items))
 
-    # An observing pointer to the C++ ListNode
-    cdef cppListNode* ptr
+    def upper_bound(self):
+        """The highest value the integer(s) can take (inclusive)."""
+        return int(self.ptr.upper_bound())
 
+    # An observing pointer to the C++ IntegerNode
+    cdef cppIntegerNode* ptr
 
-cdef class IntegerVariable(ArraySymbol):
-    """Integer decision-variable symbol.
+
+cdef class LessEqual(ArraySymbol):
+    """Smaller-or-equal comparison element-wise between two symbols.
     
     Examples:
-        This example adds an integer symbol to a model.
+        This example creates an inequality operation between integer symbols.
         
         >>> from dwave.optimization.model import Model
         >>> model = Model()
         >>> i = model.integer(25, upper_bound=100)
-        >>> type(i)
-        <class 'dwave.optimization.symbols.IntegerVariable'>
+        >>> j = model.integer(25, lower_bound=-100)
+        >>> k = i <= j
+        >>> type(k)
+        <class 'dwave.optimization.symbols.LessEqual'>
     """
-    def __init__(self, Model model, shape=None, lower_bound=None, upper_bound=None):
-        cdef vector[Py_ssize_t] vshape = _as_cppshape(tuple() if shape is None else shape )
+    def __init__(self, ArraySymbol lhs, ArraySymbol rhs):
+        if lhs.model is not rhs.model:
+            raise ValueError("lhs and rhs do not share the same underlying model")
```

Comment:
```suggestion
            raise ValueError("left and right sides do not share the same underlying model")
```

Reply:
`lhs` and `rhs` are the parameter names.